src/compile/data/calculate.ts

```diff
@@ -15,6 +18,32 @@ export class CalculateNode extends DataFlowNode {
     super();
   }
 
+  public static makeAllFromSort(model: ModelWithField): CalculateNode[] {
```

Comment:
The name is quite vague for its very specific purpose. ("Make all" of what?   This is actually a special purposed calculate node, not a generic calculate node.) 

Perhaps,  `makeAllForSortIndex` is better? 

(Also feel free to suggest better name.)

Reply:
I also start wondering that this is quite a special purposed one, so I wonder if having a distinct type of node for this will help with optimization in longer term? 
(But since this is already working, maybe it's waste of time to change it now -- I'm just asking.) 

@domoritz  -- any thoughts?

Reply:
I don't think it is that special and I can't think of a special optimization that we would do. Let's leave it for now.